x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/common/notifications/AbstractAuditor.java

```diff
@@ -83,4 +168,21 @@ private XContentBuilder toXContentBuilder(ToXContent toXContent) {
             throw new RuntimeException(e);
         }
     }
+
+    private void writeBacklog() {
+        BulkRequest bulkRequest = new BulkRequest();
+        ToXContent doc = backlog.poll();
+        while (doc != null) {
+            bulkRequest.add(indexRequest(doc));
+            doc = backlog.poll();
+        }
+
+        client.bulk(bulkRequest, ActionListener.wrap(
```

Comment:
The retrying bulk persister (`ResultsPersisterService`) would be useful here but is not accessible in this package.

#62779 fixes that by moving the persister to core but there is some debate about exactly what we should do with `ResultsPersisterService` and it probably isn't necessary here.